### PR TITLE
Namespace admin and delete user

### PR DESCRIPTION
**Pull request by HUSTToanLH**

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class Admin::UsersController < ApplicationController
+  before_action :logged_in_user
+  before_action :find_user, except: [:index, :new, :create]
```
Comment (hellovietnam93) [inline app/controllers/admin/users_controller.rb]: What is the difference between admin and normal user ?

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  def index
+    @users = User.paginate page: params[:page]
+  end
+
+  def destroy
+    unless current_user? @user
+      @user.destroy
+      flash_for_destroy @user
+    else
+      flash[:success] = t "admin.flash.user_cannot"
```
Comment (vuducluan) [inline app/controllers/admin/users_controller.rb]: flash[:danger] or flash[:warning]

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  def destroy
+    unless current_user? @user
+      @user.destroy
+      flash_for_destroy @user
+    else
+      flash[:warning] = t "admin.flash.user_cannot"
+    end
+      redirect_to admin_users_path
```
Comment (hellovietnam93) [inline app/controllers/admin/users_controller.rb]: ident

File: app/controllers/admin/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+class Admin::UsersController < ApplicationController
+  before_action :logged_in_user
+  before_action :find_user, except: [:index, :new, :create]
+  before_action :verify_admin
```
Comment (hellovietnam93) [inline app/controllers/admin/users_controller.rb]: this action should before find_user

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -19,4 +19,13 @@ def destroy
     log_out
     redirect_to root_url
   end
+
+  private
+  def redirect_with_role
+    if current_user.admin?
```
Comment (hellovietnam93) [inline app/controllers/sessions_controller.rb]: redirect_to current_user.admin? ? admin_root_path : root_url

File: app/views/admin/users/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<% provide :title, t("views.all_users") %>
+<h1><%= t "views.all_users" %></h1>
+
+<%= will_paginate @users%>
```
Comment (hellovietnam93) [inline app/views/admin/users/index.html.erb]: add space before %>

File: app/views/relationships/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  <div class="col-md-8">
+    <h3><%= @title %></h3>
+    <% if @users.any? %>
+      <ul class="users follow">
+        <%= render @users %>
+      </ul>
+      <%= will_paginate @users%>
+    <% end %>
+  </div>
+</div>
```
Comment (hellovietnam93) [inline app/views/relationships/index.html.erb]: lack of endfile

File: app/views/shared/_stats.html.erb [hunk 8]
```diff
@@ -1,12 +1,12 @@
 <% @user ||= current_user %>
 <div class="stats">
-  <a href="<%= following_user_path @user %>" >
+  <a href="<%= user_relationship_path(@user, relationship: "following") %>" >
```
Comment (hellovietnam93) [inline app/views/shared/_stats.html.erb]: should change relationship to type

**Pull request merged**